common/src/main/java/dev/lavalink/youtube/ClientException.java

```diff
@@ -0,0 +1,14 @@
+package dev.lavalink.youtube;
+
+import dev.lavalink.youtube.clients.skeleton.Client;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Wraps an exception with client context
+ */
+public class ClientException extends Exception {
+    public ClientException(@NotNull String message, @NotNull Client client, @NotNull Throwable cause) {
```

Comment:
Store `client` and `cause` outside of the string and add getters so that calling code can access these variables and choose how to proceed, if desired.